src/SqsPoller/SqsPollerHostedService.cs

```diff
@@ -62,6 +63,7 @@ private async Task Handle(string queueUrl, CancellationToken cancellationToken)
                 _logger.LogTrace(
                     "Start processing the message with id {message_id} and ReceiptHandle {receipt_handle}");
 
+                await semaphore.WaitAsync(cancellationToken);
                 await _consumerResolver
```

Comment:
Here you create a chain of tasks by using `ContinueWith` that works like: A -> B(A) -> C(B)-> D(C) -> E(D). As result you `await` task E. That probably will lead to one by one scheduling. Correct me If I'm wrong.

Reply:
Also with async/await we lost simultaneous handling of events. So in that case I think semaphore is useless